pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -294,12 +307,16 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		metrics:                     options.Metrics,
 		syncInterval:                options.GetSyncInterval(),
 		informerInit:                atomic.NewBool(false),
+		beginInitialSync:            atomic.NewBool(false),
 		initialSync:                 atomic.NewBool(false),
 		syncTimeout:                 options.SyncTimeout,
 		discoveryNamespacesFilter:   options.DiscoveryNamespacesFilter,
 		systemNamespace:             options.SystemNamespace,
 	}
-
+	// The queue requires a time duration for a retry delay after a handler error
+	c.queue = wrappedQueue{queue.NewQueue(1 * time.Second), func() bool {
+		return c.beginInitialSync.Load()
```

Comment:
Is the requiring *only* that the event is in the informer cache, or that its in the informer cache *AND* been processed? We have some internal indexes like `servicesMap` and pod mapping that will not be up to date?

Reply:
We won't actually run the queue until `initialSync` is loaded. We just start enqueueing while `SyncAll` is running. 

I'm a bit concerned about a scenario where the "ignore" condition holds and we completely drop some events not captured in `SyncAll`. I don't _think_ it's possible, if the informer indexer List calls are populated from the same source as the events.

Reply:
Right, before we sync all, ignore all the events.

> I don't think it's possible, if the informer indexer List calls are populated from the same source as the events.
Yes, could not skip any resource event